MoltenVK/MoltenVK/GPUObjects/MVKRenderPass.mm

```diff
@@ -637,28 +622,46 @@ static uint32_t getNextViewMaskGroup(uint32_t viewMask, uint32_t* startView, uin
     if ( _renderPass->getDevice()->_pMetalFeatures->deferredStoreActions ) {
         mtlAttDesc.storeAction = MTLStoreActionUnknown;
     } else {
-        mtlAttDesc.storeAction = getMTLStoreAction(subpass, isRenderingEntireAttachment, isMemorylessAttachment, hasResolveAttachment, canResolveFormat, isStencil, false);
+		// For a combined depth-stencil format in an attachment with VK_IMAGE_ASPECT_STENCIL_BIT,
+		// the attachment format may have been swizzled to a stencil-only format. In this case,
+		// we want to guard against an attempt to store the non-existent depth component.
+		MTLPixelFormat mtlFmt = attachment->getMTLPixelFormat();
+		MVKPixelFormats* pixFmts = _renderPass->getPixelFormats();
+		bool isDepthFormat = pixFmts->isDepthFormat(mtlFmt);
+		bool isStencilFormat = pixFmts->isStencilFormat(mtlFmt);
+		if (isStencilFormat && !isStencil && !isDepthFormat) {
+			mtlAttDesc.storeAction = MTLStoreActionDontCare;
+		} else {
+			mtlAttDesc.storeAction = getMTLStoreAction(subpass, isRenderingEntireAttachment, isMemorylessAttachment, hasResolveAttachment, canResolveFormat, isStencil, false);
+		}
     }
     return (mtlLA == MTLLoadActionClear);
 }
 
 void MVKRenderPassAttachment::encodeStoreAction(MVKCommandEncoder* cmdEncoder,
                                                 MVKRenderSubpass* subpass,
+												MVKImageView* attachment,
                                                 bool isRenderingEntireAttachment,
-												bool isMemorylessAttachment,
 												bool hasResolveAttachment,
 												bool canResolveFormat,
                                                 uint32_t caIdx,
                                                 bool isStencil,
                                                 bool storeOverride) {
-    MTLStoreAction storeAction = getMTLStoreAction(subpass, isRenderingEntireAttachment, isMemorylessAttachment, hasResolveAttachment, canResolveFormat, isStencil, storeOverride);
-    MVKPixelFormats* pixFmts = _renderPass->getPixelFormats();
-
-	MTLPixelFormat mtlFmt = pixFmts->getMTLPixelFormat(_info.format);
+	// For a combined depth-stencil format in an attachment with VK_IMAGE_ASPECT_STENCIL_BIT,
+	// the attachment format may have been swizzled to a stencil-only format. In this case,
+	// we want to guard against an attempt to store the non-existent depth component.
+	MTLPixelFormat mtlFmt = attachment->getMTLPixelFormat();
+	MVKPixelFormats* pixFmts = _renderPass->getPixelFormats();
 	bool isDepthFormat = pixFmts->isDepthFormat(mtlFmt);
 	bool isStencilFormat = pixFmts->isStencilFormat(mtlFmt);
 	bool isColorFormat = !(isDepthFormat || isStencilFormat);
 
+	bool isMemorylessAttachment = false;
+#if MVK_APPLE_SILICON
+	isMemorylessAttachment = attachment->getMTLTexture().storageMode == MTLStorageModeMemoryless;
+#endif
+	MTLStoreAction storeAction = getMTLStoreAction(subpass, isRenderingEntireAttachment, isMemorylessAttachment, hasResolveAttachment, canResolveFormat, isStencil, storeOverride);
```

Comment:
Wait, where's the part where we set the depth store action to `DontCare`?

In fact, if the DS image only has a stencil component, why are we telling Metal it's a depth attachment at all?

Reply:
> Wait, where's the part where we set the depth store action to DontCare?

We don't. On a stencil-only `MVKImageView` attachment, `isDepthFormat` is `false`, and `setDepthStoreAction:` will not be called at all, leaving the store action as `Unknown`

The Metal error was not a validation error, but an `EXC_BAD_ACCESS` crash on the call to `setDepthStoreAction:`, so I thought it would be best to avoid calling it altogether, in case `setDepthStoreAction:` is blindly stomping on something that was not there. Metal seems happy to leave the store action at `Unknown` if there is not a component to do something with.

>In fact, if the DS image only has a stencil component, why are we telling Metal it's a depth attachment at all?

The _image_ has both components (as does the renderpass attachment). The _image view_ only has a stencil component, based on `VK_IMAGE_ASPECT_STENCIL_BIT`. The problem was that we were previously testing the renderpass attachment format, not the image view format.

Reply:
I _could_ change this to extract a `DontCare` and call `setDepthStoreAction:` with it, which would be consistent with the related behavior in `populateMTLRenderPassAttachmentDescriptor()`, assuming `setDepthStoreAction:` didn't choke on the call.

Reply:
> > In fact, if the DS image only has a stencil component, why are we telling Metal it's a depth attachment at all?
> 
> The _image_ has both components (as does the renderpass attachment). The _image view_ only has a stencil component, based on `VK_IMAGE_ASPECT_STENCIL_BIT`. The problem was that we were previously testing the renderpass attachment format, not the image view format.

Ah, OK.

Reply:
> I _could_ change this to extract a `DontCare` and call `setDepthStoreAction:` with it, which would be consistent with the related behavior in `populateMTLRenderPassAttachmentDescriptor()`, assuming `setDepthStoreAction:` didn't choke on the call.

No, it's fine.